### PR TITLE
{devel,lib,system,tools}[GCCcore/12.3.0] IPython v8.14.0, ZeroMQ v4.3.4, libsodium v1.0.18, ...

### PR DESCRIPTION
**Pull request by casparvl**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/i/IPython/IPython-8.14.0-GCCcore-12.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ Flexible, embeddable interpreters to load into your own projects.
+ Easy to use, high performance tools for parallel computing."""
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+builddependencies = [
+    ('binutils', '2.40'),
+]
+dependencies = [
+    ('Python', '3.11.3'),
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/i/IPython/IPython-8.14.0-GCCcore-12.3.0.eb:20]:
```suggestion
    ('Python', '3.11.3'),
    ('Python-bundle-PyPI', '2023.06'),
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]
To fix the missing python packages, I would suggest adding `Python-bundle-PyPI` to the dependencies.
```
ipython 8.14.0 requires decorator, which is not installed.
ipython 8.14.0 requires pexpect, which is not installed.
ipython 8.14.0 requires pygments, which is not installed.
```

File: easybuild/easyconfigs/i/IPython/IPython-8.14.0-GCCcore-12.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    ('asttokens', '2.2.1', {'checksums': ['4622110b2a6f30b77e1473affaa97e711bc2f07d3f10848420ff1898edbe94f3']}),
+    ('stack_data', '0.6.2', {'checksums': ['32d2dd0376772d01b6cb9fc996f3c8b57a357089dec328ed4b6553d037eaf815']}),
+    ('prompt_toolkit', '3.0.39', {'checksums': ['04505ade687dc26dc4284b1ad19a83be2f2afe83e7a828ace0c72f3a1df72aac']}),
+    ('pickleshare', '0.7.5', {'checksums': ['87683d47965c1da65cdacaf31c8441d12b8044cdec9aca500cd78fc2c683afca']}),
+    ('matplotlib-inline', '0.1.6', {'modulename': False, 'checksums': ['f887e5f10ba98e8d2b150ddcf4702c1e5f8b3a20005eb0f74bfdbd360ee6f304']}),
+    ('parso', '0.8.3', {'checksums': ['8c07be290bb59f03588915921e29e8a50002acaf2cdc5fa0e0114f91709fafa0']}),
+    ('jedi', '0.18.2', {'checksums': ['bae794c30d07f6d910d32a7048af09b5a39ed740918da923c6b780790ebac612']}),
+    ('backcall', '0.2.0', {'checksums': ['5cbdbf27be5e7cfadb448baf0aa95508f91f2bbc6c6437cd9cd06e2a4c215e1e']}),
+    ('ipython', '8.14.0', {'modulename': 'IPython', 'checksums': ['1d197b907b6ba441b692c48cf2a3a2de280dc0ac91a3405b39349a50272ca0a1']}),
+]
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/i/IPython/IPython-8.14.0-GCCcore-12.3.0.eb:47]: Suggestion to make the check happy and update `jedi` to `0.19.0`:
```suggestion
exts_list = [
    ('traitlets', '5.9.0', {
        'checksums': ['f6cde21a9c68cf756af02035f72d5a723bf607e862e7be33ece505abf4a3bad9'],
    }),
    ('pure_eval', '0.2.2', {
        'checksums': ['2b45320af6dfaa1750f543d714b6d1c520a1688dec6fd24d339063ce0aaa9ac3'],
    }),
    ('executing', '1.2.0', {
        'checksums': ['19da64c18d2d851112f09c287f8d3dbbdf725ab0e569077efb6cdcbd3497c107'],
    }),
    ('asttokens', '2.2.1', {
        'checksums': ['4622110b2a6f30b77e1473affaa97e711bc2f07d3f10848420ff1898edbe94f3'],
    }),
    ('stack_data', '0.6.2', {
        'checksums': ['32d2dd0376772d01b6cb9fc996f3c8b57a357089dec328ed4b6553d037eaf815'],
    }),
    ('prompt_toolkit', '3.0.39', {
        'checksums': ['04505ade687dc26dc4284b1ad19a83be2f2afe83e7a828ace0c72f3a1df72aac'],
    }),
    ('pickleshare', '0.7.5', {
        'checksums': ['87683d47965c1da65cdacaf31c8441d12b8044cdec9aca500cd78fc2c683afca'],
    }),
    ('matplotlib-inline', '0.1.6', {
        'modulename': False,
        'checksums': ['f887e5f10ba98e8d2b150ddcf4702c1e5f8b3a20005eb0f74bfdbd360ee6f304'],
    }),
    ('parso', '0.8.3', {
        'checksums': ['8c07be290bb59f03588915921e29e8a50002acaf2cdc5fa0e0114f91709fafa0'],
    }),
    ('jedi', '0.19.0', {
        'checksums': ['bcf9894f1753969cbac8022a8c2eaee06bfa3724e4192470aaffe7eb6272b0c4'],
    }),
    ('backcall', '0.2.0', {
        'checksums': ['5cbdbf27be5e7cfadb448baf0aa95508f91f2bbc6c6437cd9cd06e2a4c215e1e'],
    }),
    ('ipython', version, {
        'modulename': 'IPython',
        'checksums': ['1d197b907b6ba441b692c48cf2a3a2de280dc0ac91a3405b39349a50272ca0a1'],
    }),
]
```

File: easybuild/easyconfigs/i/IPython/IPython-8.14.0-GCCcore-12.3.0.eb (hunk 1 above)
Comment (casparvl) [reply easybuild/easyconfigs/i/IPython/IPython-8.14.0-GCCcore-12.3.0.eb:20]: Done

File: easybuild/easyconfigs/i/IPython/IPython-8.14.0-GCCcore-12.3.0.eb (hunk 2 above)
Comment (casparvl) [reply easybuild/easyconfigs/i/IPython/IPython-8.14.0-GCCcore-12.3.0.eb]: Done

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
Test report by @casparvl
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/2a7f30d0befc9ff61fb9c82dae83fc35 for a full test report.

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
@boegelbot please test @ jsc-zen2

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/f2fa55e6efcae2b63b08fb9a8a4ae1b9 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
zen2-ubuntu-2304 - Linux Ubuntu 23.04 (Lunar Lobster), x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.11.2
See https://gist.github.com/SebastianAchilles/4f97eec46f5a73178d657e164e693693 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.9.16
See https://gist.github.com/SebastianAchilles/e00479d7622bb01095a705d4e4fb0040 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
zen2-rockylinux-88 - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/SebastianAchilles/dfc7e521e9758c84b2b456d070b18710 for a full test report.

File: easybuild/easyconfigs/l/libsodium/libsodium-1.0.18-GCCcore-12.3.0.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+sources = [SOURCE_TAR_GZ]
+checksums = ['6f504490b342a4f8a4c4a02fc9b866cbef8622d5df4e5452b46be121e46636c1']
+
+builddependencies = [
+    ('binutils', '2.40'),
+]
+
+
+sanity_check_paths = {
+    'files': ['include/sodium.h', 'lib/%(name)s.so', 'lib/%(name)s.a'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/l/libsodium/libsodium-1.0.18-GCCcore-12.3.0.eb:29]: I think it is better to use the template for the extension of shared libraries
```suggestion
    'files': ['include/sodium.h', 'lib/%(name)s.%s' % SHLIB_EXT, 'lib/%(name)s.a'],
```

File: easybuild/easyconfigs/o/OpenPGM/OpenPGM-5.2.122-GCCcore-12.3.0.eb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+builddependencies = [
+    ('binutils', '2.40'),
+    ('Python', '3.11.3'),
+]
+
+start_dir = 'pgm'
+
+sanity_check_paths = {
+    'files': ['lib/libpgm.so', 'lib/libpgm.a'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/o/OpenPGM/OpenPGM-5.2.122-GCCcore-12.3.0.eb:40]:
```suggestion
    'files': ['lib/libpgm.%s' % SHLIB_EXT, 'lib/libpgm.a'],
```

File: easybuild/easyconfigs/z/ZeroMQ/ZeroMQ-4.3.4-GCCcore-12.3.0.eb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    ('util-linux', '2.39'),
+]
+
+# Compialtion warnings in GCC 11, cf. https://github.com/zeromq/libzmq/issues/4178
+# Needto disable warnings as errors.
+configopts = "--with-pic --with-pgm --with-libsodium --disable-Werror"
+
+
+sanity_check_paths = {
+    'files': ['lib/libzmq.so', 'lib/libzmq.a'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/z/ZeroMQ/ZeroMQ-4.3.4-GCCcore-12.3.0.eb:36]:
```suggestion
    'files': ['lib/libzmq.%s' % SHLIB_EXT, 'lib/libzmq.a'],
```

File: easybuild/easyconfigs/l/libsodium/libsodium-1.0.18-GCCcore-12.3.0.eb (hunk 3 above)
Comment (casparvl) [reply easybuild/easyconfigs/l/libsodium/libsodium-1.0.18-GCCcore-12.3.0.eb:29]: You're totally right. Ugh, I hate that the `--try-*` functions replace these templates...

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
Hm, I think the issue is in combining a template and string replacement in the same string. I'll look at it tomorrow...

File: easybuild/easyconfigs/l/libsodium/libsodium-1.0.18-GCCcore-12.3.0.eb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+sources = [SOURCE_TAR_GZ]
+checksums = ['6f504490b342a4f8a4c4a02fc9b866cbef8622d5df4e5452b46be121e46636c1']
+
+builddependencies = [
+    ('binutils', '2.40'),
+]
+
+
+sanity_check_paths = {
+    'files': ['include/sodium.h', 'lib/%(name)s.%s' % SHLIB_EXT, 'lib/%(name)s.a'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/l/libsodium/libsodium-1.0.18-GCCcore-12.3.0.eb:29]: I am really sorry for make a wrong suggestion earlier. This should be the correct syntax:
```suggestion
    'files': ['include/sodium.h', 'lib/%%(name)s.%s' % SHLIB_EXT, 'lib/%(name)s.a'],
```

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/10b77063e51971bcd93229609aed44c6 for a full test report.

**Comment by casparvl** [comment]
@boegelbot please test @ jsc-zen2

**Review by SebastianAchilles — APPROVED** [review]
Thanks a lot for addressing all the comments @casparvl 

This PR looks good to me.

**Comment by SebastianAchilles** [comment]
Going in, thanks @casparvl!

**Pull request merged**